File: admin-web/src/types/building.ts
import type { PageQuery } from './common';

export interface BuildingVO {
  id: number;
  name: string;
  cover?: string;
  bannerImages?: string;
  developer?: string;
  address: string;
  description?: string;
  openingTime?: string;
  deliveryTime?: string;
  status: number;
  createTime?: string;
}

export interface BuildingPageQuery extends PageQuery {
  name?: string;
  status?: number | '';
}

export interface BuildingForm {
  id?: number;
  name: string;
  cover?: string;
  bannerImages?: string;
  developer?: string;
  address: string;
  description?: string;
  openingTime?: string;
  deliveryTime?: string;
  status: number;
}

export interface BuildingStatusUpdate {
  id: number;
  status: number;
}

File: backend/smart-sales-control/src/main/java/com/sellinghouses/salescontrol/module/building/entity/Building.java
package com.sellinghouses.salescontrol.module.building.entity;

import java.time.LocalDateTime;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Building {

    private Long id;

    private String name;

    private String cover;

    private String bannerImages;

    private String developer;

    private String address;

    private String description;

    private LocalDateTime openingTime;

    private LocalDateTime deliveryTime;

    private Integer status;

    private Integer deleted;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;

    private Long createUser;

    private Long updateUser;
}
